cardano-api/src/Cardano/Api/Typed.hs

```diff
@@ -1565,6 +1561,114 @@ data MultiSigScript = RequireSignature (Hash PaymentKey)
                     | RequireMOf Int [MultiSigScript]
   deriving (Eq, Show)
 
+instance ToJSON MultiSigScript where
+  toJSON (RequireSignature payKeyHash) = String . Text.decodeUtf8 $ serialiseToRawBytesHex $ payKeyHash
+  toJSON (RequireAnyOf reqSigs) = object [ "any" .= map toJSON reqSigs ]
+  toJSON (RequireAllOf reqSigs) = object [ "all" .= map toJSON reqSigs ]
+  toJSON (RequireMOf reqNum reqSigs) = toJSONmOfnChecks reqSigs reqNum (length reqSigs)
+
+toJSONmOfnChecks :: [MultiSigScript] -> Int -> Int -> Value
+toJSONmOfnChecks keys required total
+  | length keys /= total = error $ "The number of payment key hashes submitted \
+                                 \does not equal total. " ++ " total: " ++ show total
+                                 ++ "Number of key hashes: " ++ show (length keys)
+
+  | length keys < required = error $ "required exceeds the number of payment key \
+                                   \hashes. Number of keys: " ++ show (length keys)
+                                   ++ " required: " ++ show required
+
+  | required <= 0 = error $ "setting required to 0 is nonsensiscal. Please refer to the docs."
```

Comment:
Nitpicking about spelling as usual :smile:

```suggestion
  | required <= 0 = error $ "setting required to 0 is nonsensical. Please refer to the docs."
```

An alternative error message might be: `The required number of payment key hashes cannot be less than or equal to 0`.

Reply:
:+1: Will fix in a follow up PR